stratum/hal/lib/common/common.proto

```diff
@@ -250,6 +252,59 @@ enum FecMode {
   FEC_MODE_AUTO = 3;
 }
 
+// The input optical power of this port in units of 0.01 decibel-milliwatt.
+// If the port is an aggregate of multiple physical channels, this attribute
+// is the total power or sum of all channels.
+//
+// The time-based values are the timestamps in nanoseconds relative to the
+// Unix Epoch (Jan 1, 1970 00:00:00 UTC)."
+message InputPower {
+  // The instantaneous value of the statistic.
+  ywrapper.Decimal64Value instant = 1;
+  // The arithmetic mean value of the statistic over the time interval.
+  ywrapper.Decimal64Value avg = 2;
+  // The time interval over which the statistics are computed.
+  uint64 interval = 3;
+  // The maximum value of the statistic over the time interval.
+  ywrapper.Decimal64Value max = 4;
+  // The absolute time at which the maximum value occurred.
+  uint64 max_time = 5;
+  // The minimum value of the statistic over the time interval.
+  ywrapper.Decimal64Value min = 6;
+  // The absolute time at which the minimum value occurred.
+  uint64 min_time = 7;
+}
+
+// Target output optical power level of the optical channel, expressed in
+// increments of 0.01 decibel-milliwats.
+//
+// The time-based values are the timestamps in nanoseconds relative to the
+// Unix Epoch (Jan 1, 1970 00:00:00 UTC)."
+message OutputPower {
+  // The instantaneous value of the statistic.
+  ywrapper.Decimal64Value instant = 1;
+  // The arithmetic mean value of the statistic over the time interval.
+  ywrapper.Decimal64Value avg = 2;
+  // The time interval over which the statistics are computed.
+  uint64 interval = 3;
+  // The maximum value of the statistic over the time interval.
+  ywrapper.Decimal64Value max = 4;
+  // The absolute time at which the maximum value occurred.
+  uint64 max_time = 5;
+  // The minimum value of the statistic over the time interval.
+  ywrapper.Decimal64Value min = 6;
+  // The absolute time at which the minimum value occurred.
+  uint64 min_time = 7;
+}
+
+// NOTE. The name 'LazerFrequency' was chosen to avoid the conflict with
+// 'gnmi_publisher.cc:Frequency'.
+//
+// Optical channel frequency in MHz.
+message LazerFrequency {
```

Comment:
Do we mean Laser instead of Lazer? Or, does "lazer" have a special meaning in TAI?

Reply:
Nothing special for TAI case, just a spelling mistake. We mean "laser" here.
Fixed.